metacat-thrift/src/main/java/com/netflix/metacat/thrift/HiveConvertersImpl.java

```diff
@@ -331,16 +328,25 @@ public PartitionDto hiveToMetacatPartition(final TableDto tableDto, final Partit
         } catch (MetaException e) {
             throw new IllegalArgumentException("Invalid partition name", e);
         }
-
-        final List<String> partVals = Lists.newArrayList();
-        for (String key : tableDto.getPartition_keys()) {
-            final String val = hm.get(key);
-            if (val == null) {
-                throw new IllegalArgumentException("Invalid partition name - missing " + key);
+        // Get the partition keys.
+        List<String> partitionKeys = null;
+        if (tableDto != null) {
+            partitionKeys = tableDto.getPartition_keys();
+        }
+        // If table has not been provided, return the values without validating.
+        if (partitionKeys != null) {
+            final List<String> partVals = Lists.newArrayListWithCapacity(partitionKeys.size());
+            for (String key : partitionKeys) {
```

Comment:
Since we allow the non-validation case, Why do we need the validation? The reason to ask is how come we can get partitions that invalid into the system. If that's not possible, we don't need to worry about this.

Reply:
I did not remove it since the logic was there already. We need to revisit it when we make the second part of the changes.